src/amp-story-player/amp-story-player-impl.js

```diff
@@ -1317,14 +1317,16 @@ export class AmpStoryPlayer {
   maybeGetCacheUrl_(url) {
     const ampCache = this.element_.getAttribute('amp-cache');
 
-    if (
-      !ampCache ||
-      isProxyOrigin(url) ||
-      !SUPPORTED_CACHES.includes(ampCache)
-    ) {
+    if (!ampCache || !SUPPORTED_CACHES.includes(ampCache)) {
       return Promise.resolve(url);
     }
 
+    if (isProxyOrigin(url)) {
+      // Ensures serving type is 'viewer' (/v/) when publishers provide their
+      // own cached URL.
+      return Promise.resolve(url.replace('/c/', '/v/'));
```

Comment:
Is it possible that the url doesn't include `amp_js_v`? If so, then forcing it to `/v/` will break it, because the cache today treats `/v/` without an `amp_js_v` as a 404 error.

Reply:
> Could this match part of the publisher's own url, instead of just the proxy's URL?

Right, let's be more specific here and check for the first 3 chars of the path

> Is it possible that the url doesn't include amp_js_v?

the `amp_js_v` is always added later in the code execution, when we add parameters for the AMP document such as `cap` etc

Reply:
Checking the first characters sounds good to me.

Reply:
Done.